### DISCUSSION TITLE
Over temperature power measurement cut off

### DISCUSSION THREAD

**Discussion by blacknell** [arendst/Tasmota#20516] [Q&A]
I have a Sonoff POWR320D Elite and after diagnosis I can see that the firmware switched all power off after detecting a core temperature of 91.1C

```
Jan 11 09:05:34 tasmota-ED72EC-4844 ESP-NRG: Temperature 91.1
Jan 11 09:05:34 tasmota-ED72EC-4844 ESP-SRC: Overtemp
```

I've seen many posts everywhere saying do not rely on the ESP32 core temperature as it's completely inaccurate so I wondering whether this is real.

I have 5 such devices doing the same job in different rooms. With power off and in an steady state (ambient temperature in all units is 12C), 3 of them read ESP32 core between 35-41C, one reads 22C. The problematic one above reads 62C.

At the time of the cutoff above the ambient temperature near that unit was 39C

My question, if the ESP32 core temperature can't be relied upon then should we really use it as a cutoff? I have rules that cut power when ambient gets to 50C which it never has done in stress testing.

Second part to the question, is this particular device likely to be faulty (or about to give up the ghost) given its reading 62C at 12C ambient?

I'd like to believe the reading is completely unreliable and to switch off the overtemp trip completely with `SetSensor127 0` But it makes me uneasy.

See [related discussion](https://github.com/arendst/Tasmota/discussions/15598)

**Comment by Wintespe** [comment]
If you have 5 devices, I would simply measure the surface temperature of the ESP - if possible - with a non-contact thermometer and compare them with each other. You will then get an indication of the temperature behaviour.
